hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSink.java

```diff
@@ -190,6 +190,7 @@ public void replicateEntries(List<WALEntry> entries, final CellScanner cells,
             for (int i = 0; i < count; i++) {
               // Throw index out of bounds if our cell count is off
               if (!cells.advance()) {
+                // TODO(baugenreich) increment failed sink?
```

Comment:
I think we should be incrementing the failure metric here since this runtime exception would not be caught ?

Reply:
it might be better to add a new catch clause to this wrapping try. We can just re-throw the exception after incrementing. I wonder if it should be logged too. Either that or is there any error handling in whatever calls this method?